Source/KakapoServer.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  KakapoServer.swift
+//  KakapoExample
+//
+//  Created by Joan Romano on 31/03/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+class KakapoServer: NSURLProtocol {
+    
+    typealias KakapoLookupObject = (method: KakapoHTTPMethod,
+                                    handler: (request: KakapoRequest) -> ())
+    
+    enum KakapoHTTPMethod: String {
+        case GET, POST, PUT, DELETE
+    }
+    
+    struct KakapoRequest {
+        let urlString: String
+        let info: URLInfo
+    }
+    
+    private static var routes: [String : KakapoLookupObject] = [:]
+    
+    class func enable() {
+        NSURLProtocol.registerClass(self)
```

Comment:
I guess it should have no effects? Not sure if we should handle that failure case in NSURLProtocol (if any) but could be an option.